src/Umbraco.Core/Persistence/SqlSyntax/MySqlSyntaxProvider.cs

```diff
@@ -277,7 +277,7 @@ public override string Format(IndexDefinition index)
 
         public override string Format(ForeignKeyDefinition foreignKey)
         {
-            return string.Format(CreateForeignKeyConstraint,
+            return string.Format(Format(foreignKey),
```

Comment:
I don't think this is going to work, this is calling itself recursively.. in an infinite loop.

I think it's fine to remove the method from the interface but not from the `MySqlSyntaxProvider` itself. I'll update.